File: Algorithms/src/1460. Make Two Arrays Equal by Reversing Subarrays/Code/solution.cpp
#include <vector>
#include <algorithm> // For std::sort

class Solution
{
public:
    bool canBeEqual(std::vector<int> &target, std::vector<int> &arr)
    {
        // Step 1: Sort the 'target' array
        // Sorting helps in comparing the arrays element by element in a straightforward manner
        std::sort(target.begin(), target.end());

        // Step 2: Sort the 'arr' array
        // After sorting both arrays, they should have the same elements in the same order if they can be made equal
        std::sort(arr.begin(), arr.end());

        // Step 3: Compare the sorted arrays
        // If the sorted arrays are identical, it means the original arrays can be made equal by reordering
        return target == arr;
    }
};

File: Algorithms/src/1460. Make Two Arrays Equal by Reversing Subarrays/Code/solution.go
import "sort"

// canBeEqual checks if the target array can be made equal to the arr array
// by sorting both arrays and comparing their elements.
func canBeEqual(target []int, arr []int) bool {
    // Step 1: Sort the target array in ascending order.
    sort.Ints(target)
    // Step 2: Sort the arr array in ascending order.
    sort.Ints(arr)
    
    // Step 3: Compare each element of the sorted target array with the sorted arr array.
    for i := range target {
        // Step 4: If any corresponding elements are not equal, return false.
        if target[i] != arr[i] {
            return false
        }
    }
    // Step 5: If all elements match, return true.
    return true
}
